js/components/dataSelector.js

```diff
@@ -1,32 +1,79 @@
 var React = require('react'),
     ClassNames = require('classnames'),
-    Icon = require('./icon');
+    AccessibleButton = require('./accessibleButton'),
+    AccessibleMenu = require('./higher-order/accessibleMenu'),
+    Icon = require('./icon'),
+    CONSTANTS = require('../constants/constants');
 
 var DataSelector = React.createClass({
+
   getInitialState: function() {
     return {
       currentPage: 1
     };
   },
 
-  handleDataSelection: function(dataItem) {
+  componentWillMount: function() {
+    this.leftChevronBtn = null;
+    this.rightChevronBtn = null;
+    this.itemButtons = {};
+    this.autoFocus = {
+      first: false,
+      last: false,
```

Comment:
What are first and last used for?

Reply:
You can auto focus the first, last or currently selected elements depending on the situation. First and last are used to focus on an element when switching to a different page with the left or right arrows.